### PR TITLE
Feature/bs 1.19

### PR DESCRIPTION
**Pull request by ErisApps**
Fixes compatibility for 1.19
Also migrates from the BS-Utils config system to the BSIPA config system

### CODE REVIEW

File: UI/ProgressBar.cs [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
 using TMPro;
 using UnityEngine;
 using UnityEngine.SceneManagement;
 using UnityEngine.UI;
 
-namespace SongCore
+namespace SongCore.UI
 {
-    public class ProgressBar : MonoBehaviour
+    internal class ProgressBar : MonoBehaviour
```
Comment (rithik-b) [inline UI/ProgressBar.cs:12]: Please make this public, PlaylistManager uses it

**Comment by ggggggggggggggggggggglo** [comment]
cool

File: Plugin.cs [hunk 2]
```diff
@@ -33,30 +32,39 @@ public class Plugin
         public static string noArrowsCharacteristicName = "NoArrows";
 
         [Init]
-        public void Init(IPALogger pluginLogger)
+        public void Init([Config.Name(nameof(SongCore) + "/" + nameof(SongCore))]Config conf, IPALogger pluginLogger)
```
Comment (ErisApps) [inline Plugin.cs:35]: This apparently breaks the game when the SongCore sub folder isn't present.

Comment (Meivyn) [reply Plugin.cs:35]: Still not solved with the latest BSIPA.

Comment (ErisApps) [reply Plugin.cs:35]: Apparently pushed the changes onto a branch of my SongCore fork and not on this branch, changes have been included now.

Comment (ErisApps) [reply Plugin.cs]: After some back-and-forth, it is now truly fixed 🙂

**Comment by ErisApps** [comment]
Given that a version containing all these changes has been released and approved on Mod Assistant, I'm just gonna go ahead and merge this 🙂

**Pull request merged**